lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2467,6 +2467,170 @@ static bool checkFunctionSignature(
   return checkFunctionSignature(requiredResultFnTy, candidateFnTy.getResult());
 };
 
+// SWIFT_ENABLE_TENSORFLOW
```

Comment:
I made a best-effort attempt to clean up `TypeCheckAttr.cpp` and unify code.
More cleanup is likely possible, but I will defer that for now to unblock progress.